tests/gold_tests/pluginTest/header_rewrite/header_rewrite.test.py

```diff
@@ -22,7 +22,8 @@
 '''
 # need Curl
 Test.SkipUnless(
-    Condition.HasProgram("curl", "Curl need to be installed on system for this test to work")
+    Condition.HasProgram("curl", "Curl need to be installed on system for this test to work"),
```

Comment:
I still kinda hate this test... I mean, if the system doesn't have a curl that works, why even bother running the autest at all? For something as important as curl, we should just fail early on in autest (maybe on startup). That should be true for other critical tools that it needs.

Reply:
@dragon512 what do you think? Would it be possible to move these critical tool conditions into something like init.cli.ext or a test of its own? (Second one is probably not a great idea.)

Reply:
I agree with @zwoop, I don't see a reason for skipping tests.  It sounds like there is a problem with the development/CI environment for not having some of the dependencies installed.  Those issues should be fixed and make sure that the code base is being tested throughly.

What is the rational on skipping tests?